pkg/services/teams.go

```diff
@@ -151,7 +152,13 @@ func NewTeamsService(opts TeamsOptions) NotificationService {
 }
 
 func (s teamsService) Send(notification Notification, dest Destination) error {
-	webhookUrl, ok := s.opts.RecipientUrls[dest.Recipient]
+
+	_, err := url.ParseRequestURI(dest.Recipient)
```

Comment:
Instead of assuming that recipient is URL can we add `url`  field to the team template and let the end-user decide if recipient is the URL or something else. e.g.:

```yaml
template.app-sync-succeeded: |
  teams:
    url:  https://example.com
```


As I understand the teams URL might have sensitive information. Using `url` field in template admins might support URL "alias" e.g.:

```yaml
template.app-sync-succeeded: |
  teams:
    {{if eq .recipient "teams1"}}
    url:  https://example1.com?abc
    {{else}}{{if eq .recipient "teams2"}}
    url:  https://example2.com?abc
    {{else}}
    url:  https://example3.com?abc
    {{end}}{{end}}
```

WDYT?

Reply:
Then the problem will be exactly the same as before. The templates are placed into the config map, which isn't accessible for the people outside of the "argo management team".

Here's the real-world scenario: 
Some team is taking care of the pipelines, and providing to other dev teams the opportunity to put somewhere the JSON file with their app definition. Then this JSON is being pulled by the Argo ApplicationSets controller and placed into Argo etc. 
So, the dev team can only edit that JSON file for their needs somehow. They can place the webhook URL for teams into it easily. But in a current way, they will have to ask the pipelines team to put URL into the config map all the time to achieve this, because somebody still needs to place the webhook URL into the config map.

Reply:
> Then the problem will be exactly the same as before. 

Admin can still create template with the url that just gets the recipient value: `url: {{recipient}}`. In this case user will be able to specify the full url. I think this approach is safer because it is up to admin whether he/she wants to let end-users override the url or not.

I hope I'm not missing anything.

Reply:
@alexmt  sounds like a good trick. I will check it a bit later, but for now, it looks for me that line:
	webhookUrl, ok := s.opts.RecipientUrls[dest.Recipient]
	will prevent the sending if there's no RecipientUrls[dest.Recipient] value, which is supposed to be absent, in case when the recipient is an URL. Am I wrong?

Reply:
@alexmt  checked the way you provided in your suggestion. The thing is that in this case we still have to define recipient in the configmap, otherwise the check  `s.opts.RecipientUrls[dest.Recipient] ` will never be passed. 
So I've added a different approach: explicit enabling of the URL passing feature in the teams service options.

Reply:
@alexmt Do you have any more comments on this ?:)

Reply:
@alexmt maaaaaybe we can merge this? :)